### ISSUE TITLE
[pni] cross page: "Join us" content exists on page is invisible state

### ISSUE THREAD

**Issue by Pomax** [mozilla/foundation.mozilla.org#4638]
The "Join us" content is invisible instead of display:none, which means tabbing to the main content will "get stuck" cycling through everything in the "join us" newsletter slide-out panel, without any indication that that's where focus is.

We can either make that slide-out be `display:none` with extra work to make sure the slide out/slide in transitions work properly, or we can make it so that if there is focus anywhere inside the slide-out, the slide-out itself gets revealed (so that tabbing into it reveals it, and tabbing out of it hides it again)

**Comment by Pomax** [comment]
@kristinashu  @mmmavis my preference would be to make the newsletter content be `display:none` until it's needed, but if you think we should just auto-reveal when people tab through the content I can make that happen instead.

**Comment by kristinashu** [comment]
Sorry, what is the Join Us content you're talking about? Is it the Donate prompt that used to pop up?

**Comment by mmmavis** [comment]
@kristinashu I think Pomax was talking about the slide-down nav newsletter form 
<img width="1103" alt="image" src="https://user-images.githubusercontent.com/2896608/82858645-53d05a00-9ec9-11ea-9cbc-1f2d4779ebe5.png">

@Pomax our newsletter form (join us component) and nav code are very intertwined. This is because we use the same component and script for different scenarios (mobile, desktop, nav, footer, PNI, YouTube campaign, etc). In my opinion, the code has become pretty troublesome to deal with 😓 and since we don't have interaction tests in places for them, it's hard to spot regression when we introduce a new code change. I would say we should choose the most straightforward solution so we don't have to modify the existing code much.

> my preference would be to make the newsletter content be display:none until it's needed, but if you think we should just auto-reveal when people tab through the content I can make that happen instead.

By "display:none until it's needed", did you mean when users tab through the page, we show the nav newsletter form without any slide down effect? How is it different than the "auto-reveal" approach?

**Comment by Pomax** [comment]
The two options would be:

1. `display:none` so that the content does not exist on the page. The only way to reveal it is to click the "newsletter" button, which will flip the display property to `display:block` and then perform the slide-out animation in the same way it does right now. Dismissing it would also preserve the current animation, but set a `display:none` as final step so that content is gone, as far as the DOM is concerned.
2. We keep the current behaviour with `opacity: 0` but the moment useraction puts focus in its content, we immediately trigger the reveal animation, and when focus leaves the content, we trigger the hide animation

Of these, (2) is the most accessible but also a bit "mystery meat": there is nothing to suggest more content will pop into existence. So on that basis, I'd personally prefer (1)
